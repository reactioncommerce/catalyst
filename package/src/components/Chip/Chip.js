import React from "react";
import PropTypes from "prop-types";
import MuiChip from "@material-ui/core/Chip";
import makeStyles from "@material-ui/core/styles/makeStyles";

const useStyles = makeStyles((theme) => ({
  containedPrimary: {
    color: theme.palette.primary.contrastText,
    backgroundColor: theme.palette.colors.red
  },
  outlinedPrimary: {
    color: theme.palette.colors.red,
    border: `1px solid ${theme.palette.colors.red}`
  }
}));

/**
 * @name Chip
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */
const Chip = React.forwardRef(function Chip(props, ref) {
  const {
    color,
    ...otherProps
  } = props;

  const classes = useStyles();

  if (color === "error") {
    return (
      <MuiChip
        classes={{
          containedPrimary: classes.containedPrimary,
          outlinedPrimary: classes.outlinedPrimary
        }}
        color="primary"
        ref={ref}
        {...otherProps}
      />
    );
  }

  return (
    <MuiChip
      color={color}
      ref={ref}
      {...otherProps}
    />
  );
});

Chip.propTypes = {
  /**
   * CSS Classes
   */
  classes: PropTypes.object,
  /**
   * The color of the component
   */
  color: PropTypes.oneOf(["default", "primary", "secondary", "error"]),
  /**
   * The variant to use
   */
  variant: PropTypes.oneOf(["default", "outlined"])
};

Chip.defaultProps = {
  color: "primary",
  variant: "outlined"
};

export default Chip;
