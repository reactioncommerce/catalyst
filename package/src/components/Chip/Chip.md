### Overview

The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information. 

### Usage

#### Default Catalyst chip

This is what a chip with all the default prop options looks like:

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Chip label="Default" />
  </div>
</div>
```

It's a chip with `variant` set to `outlined`, and `color` set to `primary`.

#### Catalyst-custom chips

- **Error chip**: The error chip is used to indicate an error condition.

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Chip color="error" label="Error" />
  </div>
</div>
```

