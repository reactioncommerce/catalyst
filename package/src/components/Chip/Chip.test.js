import React from "react";
import { render } from "../../tests/index.js";
import Chip from "./Chip";

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<Chip />);
  expect(asFragment()).toMatchSnapshot();
});

test("error chip snapshot", () => {
  const { asFragment } = render(<Chip color="error" variant="contained" />);
  expect(asFragment()).toMatchSnapshot();
});
