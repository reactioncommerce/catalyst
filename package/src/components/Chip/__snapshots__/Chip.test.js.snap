// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</DocumentFragment>
`;

exports[`error chip snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary makeStyles-outlinedPrimary-34"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</DocumentFragment>
`;
